flake.nix

```diff
@@ -15,38 +15,97 @@
         flake-utils.follows = "flake-utils";
       };
     };
+    crate2nix = {
+      url = "github:nix-community/crate2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
   };
 
-  outputs = { self, nixpkgs, rust-overlay, flake-utils, solc }:
+  outputs = { self, nixpkgs, rust-overlay, flake-utils, solc, crate2nix }:
     flake-utils.lib.eachDefaultSystem (system:
       let
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [ rust-overlay.overlays.default solc.overlay ];
+          overlays = [
+            rust-overlay.overlays.default
+            solc.overlay
+            (final: prev: {
+              solc = (solc.mkDefault final final.solc_0_8_27);
+            })
+          ];
         };
         lib = pkgs.lib;
         toolchain = pkgs.rust-bin.stable.latest.default.override {
           extensions = [ "rustfmt" "clippy" "rust-src" ];
         };
+        crate2nix-tools = crate2nix.tools.${system};
+        generated = crate2nix-tools.generatedCargoNix {
+          src = ./.;
+          name = "foundry";
+        };
+        buildRustCrateForPkgs = pkgs: with pkgs; buildRustCrate.override {
+          defaultCrateOverrides = defaultCrateOverrides // {
+            chisel = attrs: {
+              # The subcrates don't get the full workspace source, so we need to
+              # update this path.
+              prePatch = (attrs.prePatch or "") + ''
+                substituteInPlace \
+                  src/session_source.rs \
+                  --replace-fail \
+                  '../../../testdata/cheats/Vm.sol' \
+                  '${./testdata/cheats/Vm.sol}'
+              '';
+            };
+            svm-rs-builds = attrs: {
+              # Why is this even a thing? It seems silly to generate the list of
+              # releases at build time.
+              # Force svm-rs-builds to generate in offline mode and see how
+              # much breaks.
+              prePatch = (attrs.prePatch or "") + ''
+                substituteInPlace \
+                  build.rs \
+                  --replace-fail \
+                  'generate();' \
+                  'generate_offline();'
+              '';
+            };
```

Comment:
This is less than ideal, is there a way you can pass in the `_offline` flag?

https://github.com/alloy-rs/svm-rs/blob/b97a212ddd72671d3f93d3b8c773ffe2edfb1ece/crates/svm-builds/build.rs#L183-L184

Either way we do control `svm-rs` and can accommodate for this usecase I think

cc @DaniPopes

Reply:
Oh cool, I can override features using crate2nix instead of having to do something ugly like plumb it up through the layers of dependencies or add a svm-rs-buids as a top-level dependency _just_ for feature activation.

It's a slight divergence from how cargo works, but definitely cleaner than the patch!